### PR TITLE
HPCC-19259  Move all spray and despray tests from hthor to thor and roxie target

### PR DESCRIPTION
**Pull request by AttilaVamos**
Change exclusion (//no<engine>) to hthor

Signed-off-by: AttilaVamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@richardkchapman please check and merge.

File: testing/regress/ecl/spray_expire_test.ecl [hunk 1]
```diff
@@ -15,19 +15,17 @@
     limitations under the License.
 ############################################################################## */
 
-import Std.File AS FileServices;
-import $.setup;
-prefix := setup.Files(false, false).FilePrefix;
-
 // This is not an engine test, but a DFU.
 // Doesn't matter much which engine does it, so we restrict to only one
```
Comment (richardkchapman) [inline testing/regress/ecl/spray_expire_test.ecl]: Should this say nothor too then?

File: testing/regress/ecl/spray_header_test.ecl [hunk 2]
```diff
@@ -15,26 +15,30 @@
     limitations under the License.
 ############################################################################## */
 
-import Std.File AS FileServices;
-import $.setup;
-prefix := setup.Files(false, false).FilePrefix;
-
 // This is not an engine test, but a DFU.
 // Doesn't matter much which engine does it, so we restrict to only one
```
Comment (richardkchapman) [inline testing/regress/ecl/spray_header_test.ecl]: Looks like it will try to run on thor too to me

File: testing/regress/ecl/spray_test.ecl [hunk 3]
```diff
@@ -15,25 +15,32 @@
     limitations under the License.
 ############################################################################## */
 
-import Std.File AS FileServices;
-import $.setup;
-prefix := setup.Files(false, false).FilePrefix;
-
 // This is not an engine test, but a DFU.
 // Doesn't matter much which engine does it, so we restrict to only one
```
Comment (richardkchapman) [inline testing/regress/ecl/spray_test.ecl]: ditto

**Comment by richardkchapman** [comment]
Is the intention here to target thor and roxie engines (separately) ? If so then there are some comments that are no longer accurate. But given I don't think any code will actually execute on thor for these tests then perhaps it makes sense to leave the nothor in ?

**Comment by AttilaVamos** [comment]
@richardkchapman Yes all spray test intend to run on both Thor and Roxie engines, based on all test cases generates their own single/multi part logical file, despray it, then test spray features.
Despray test intend to run only on Roxie, based on it tests Dropzone Restriction feature and try to despray file to illegal target and it causes exception in DFU server. 
Obsolete comments are removed. 
Please review again.

**Pull request merged**